services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationService.kt

```diff
@@ -218,16 +217,41 @@ open class ApplicationService(
         logger.info {
             "Application sent, fetching application identifier and status. id=$id, alluid=${application.alluid}."
         }
-        getApplicationInformationFromAllu(application.alluid!!)?.let {
-            application.applicationIdentifier = it.applicationId
-            application.alluStatus = it.status
+        getApplicationInformationFromAllu(application.alluid!!)?.let { response ->
+            application.applicationIdentifier = response.applicationId
+            application.alluStatus = response.status
+            initAccessForApplication(application, hanke, userId)
         }
 
         logger.info("Sent application id=$id, alluid=${application.alluid}")
         // Save only if sendApplicationToAllu didn't throw an exception
         return applicationRepository.save(application).toApplication()
     }
 
+    private fun initAccessForApplication(
+        application: ApplicationEntity,
+        hanke: HankeEntity,
+        userId: String
+    ) {
+        val hankeId = hanke.id
+        val hankeTunnus = hanke.hankeTunnus
+        val hankeNimi = hanke.nimi
+
+        if (hankeId == null || hankeTunnus == null || hankeNimi == null) {
+            throw ApplicationArgumentException(
+                "Hanke info missing: (id=$hankeId, tunnus=$hankeTunnus, nimi=$hankeNimi)"
+            )
+        }
```

Comment:
We should push to get HAI-1893 to the next sprint to avoid this kind of nonsense.